src/Compilers/CSharp/Portable/Compilation/CSharpCompilation.cs

```diff
@@ -1730,6 +1730,16 @@ bool checkValid(MethodSymbol candidate, bool isCandidate, DiagnosticBag specific
                         }
                     }
                 }
+                else if (LanguageVersion >= MessageID.IDS_FeatureAsyncMain.RequiredVersion() && taskEntryPoints.Count == 1 && viableEntryPoints.Count == 1)
+                {
+                    var info = new CSDiagnosticInfo(
+                         ErrorCode.WRN_SyncAndAsyncEntryPoints,
+                         args: Array.Empty<object>(),
+                         symbols: ImmutableArray<Symbol>.Create(viableEntryPoints.First(), taskEntryPoints.First()),
+                         additionalLocations: ImmutableArray<Location>.Create(viableEntryPoints.First().Locations.First(), taskEntryPoints.First().Locations.First());
+
+                    diagnostics.Add(new CSDiagnostic(info, viableEntryPoints.First().Locations.First()));
```

Comment:
@333fred, I updated the passed `symbols` and `additionalLocations` to include both entry points. Currently, the warning will always be added to the sync one. Should this be changed? and how?

Reply:
It doesn't need to be changed, I think.